lib/matplotlib/colorbar.py

```diff
@@ -900,6 +900,406 @@ def remove(self):
         fig.delaxes(self.ax)
 
 
+class ColorsquareBase(cm.ScalarMappable):
+    n_rasterize = 50  # rasterize solids if number of colors >= n_rasterize
+    def __init__(self, ax, cmap=None,
+                 norm=None,
+                 alpha=None,
+                 xvalues=None,
+                 yvalues=None,
+                 xboundaries=None,
+                 yboundaries=None,
+                 xticks=None,
+                 yticks=None,
+                 xformat=None,
+                 yformat=None,
+                 drawedges=False,
+                 filled=True,
+                 xlabel='',
+                 ylabel='',
+                 ):
+        #: The axes that this colorbar lives in.
```

Comment:
Unless there is a strong reason to expose them, attributes should be private (start with an underscore) by default.  Otherwise, they all become part of the public API, which is very difficult to change while maintaining back-compatibility.
(Same comment throughout the PR.)